### PR TITLE
Map zoom level fix

### LINKED ISSUE #70: Zoom level is set to the middle of the route/polyline and does not display route in its entirety
**Description:**
- Currently the camera zoom level when the polyline is added is set to the center of the polyline, which does not work well for long routes

**Steps to reproduce:**
1. Click on the Navigate button
2. Select "Search for location"
3. Enter start and end location as to be several kilometers apart (ex. Enter start as "Université de Montréal" and enter end location as "Concordia University Sir George Williams")
5. Click "Go"

**Expected:**
- Entire route is visible on the map; the map camera should zoom out to display the route in its entirety

**Actual:**
- Only a small portion of the route is visible, because the view is zoomed in too far; the map camera is centered in the middle of the route and does not display the route in its entirety

**Workaround:**
- Manually zoom out on the map to view the full polyline in its entirety

**Screenshot:**
- What happens after imputing "Cafe Myriade" and "H3Y1S4" into the start and destination respectively:

Screenshot_1583104316

### PR DESCRIPTION
**Pull request by jonZlotnik**
Fixed it!
70-directions-zoom-bugfix

### CODE REVIEW

File: app/src/main/java/com/example/campusguide/directions/Route.kt [hunk 1]
```diff
@@ -115,9 +115,9 @@ class Route constructor(private val map: Map, private val activity: FragmentActi
                     begin = map.addMarker(startPoint)
                     dest = map.addMarker(endPoint)
                     map.moveCamera(
-                        CameraUpdateFactory.newLatLngZoom(
-                            routeBounds.center,
-                            Constants.ZOOM_STREET_LVL
+                        CameraUpdateFactory.newLatLngBounds(
+                            routeBounds,
+                            100 // padding around the route in unknown units
```
Comment (jun-loke) [inline app/src/main/java/com/example/campusguide/directions/Route.kt]: Padding is in pixels
https://developers.google.com/android/reference/com/google/android/gms/maps/CameraUpdateFactory#parameters_1

Comment (felixlapierre) [reply app/src/main/java/com/example/campusguide/directions/Route.kt]: Update the comment to indicate that it's pixels before you merge

File: .gitignore [hunk 2]
```diff
@@ -18,9 +18,241 @@
...
+/app/src/main/res/values/google_maps_key.xml
 
 # Code coverage reports
 coverage
 
 # Objectbox
-app/objectbox-jni-windows-x64.dll
\ No newline at end of file
+app/objectbox-jni-windows-x64.dll
+
```
Comment (felixlapierre) [inline .gitignore]: This is stuff from another PR, make sure you merge the other PR before this one

File: README.md [hunk 3]
```diff
@@ -22,11 +22,9 @@ Create a campus guide application to help students locate their classrooms.
...
 1. Download and install [Android Studio](https://developer.android.com/studio)
-2. [Get an API key for Google Maps](https://developers.google.com/maps/documentation/android-sdk/get-api-key)
-3. Clone the repository onto your computer
-4. Create a copy of the file `app\src\debug\res\values\google_maps_api.example.xml` in the same folder.
-5. Rename that file `google_maps_api.xml`
-6. The file contains a single `<string>` tag. Change the `name` property from `google_maps_key_ex` to `google_maps_key`. Change the contents of the tag from `google_maps_key` to the value of your API key (it starts with `AIza`)
-7. Perform steps 4 through 6 with the file `app\src\release\res\values\google_maps_api.example.xml`. This step is only necessary to build the app in Release configuration.
-8. Open the project in Android Studio
-9. Run the project using an emulator or using a connected android device.
+1. [Get an API key for Google Maps](https://developers.google.com/maps/documentation/android-sdk/get-api-key)
```
Comment (felixlapierre) [inline README.md]: Again, make sure to merge the other pr first

File: app/src/main/java/com/example/campusguide/directions/ChooseDirectionOptions.kt [hunk 4]
```diff
@@ -85,10 +85,10 @@ class ChooseDirectionOptions constructor(private val route: Route?) : DialogFrag
         val getDirectionsDialogFragment =
             GetDirectionsDialogFragment(
                 GetDirectionsDialogFragment.DirectionsDialogOptions(
-                    message,
-                    startingPoint,
-                    null,
-                    EmptyDirectionsGuard(
+                    message = message,
```
Comment (felixlapierre) [inline app/src/main/java/com/example/campusguide/directions/ChooseDirectionOptions.kt:89]: Good refactoring, adds clarity :)

**Review by felixlapierre — APPROVED** [review]
Good stuff 😄 
This PR has changes from other PRs, make sure you merge the other PRs first

**Pull request merged**